### PR TITLE
Ctl 004 create contact detail screen

### PR DESCRIPTION
**Pull request by heupotato**
@lngyeen Pull Request: Screen for contact's detail

### CODE REVIEW

File: lib/pic/assets.dart [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Assets{
+  static const String placeholder = "../pic/avatar_placeholder.png";
```
Comment (lngyeen) [inline lib/pic/assets.dart]: Wrong path: change to "./pic/avatar_placeholder.png"

File: lib/ui/screens/contact_detail_screen.dart [hunk 2]
```diff
@@ -1,20 +1,120 @@
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart:1]: UI Error:  
Simulator Screen Shot - iPhone 11 Pro Max - 2021-06-23 at 16 11 30

File: lib/ui/screens/contact_detail_screen.dart [hunk 3]
```diff
@@ -1,20 +1,120 @@
...
+        color: Colors.teal[900], 
+        thickness: 3,
+        indent: 200,
+        endIndent: 200,
+    ); 
+
+
     ContactDetail({Key? key, required this.contact}) : super(key: key); 
+
+    List<Widget> appBarPic(String contactName){
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart]: Should create a dedicated StatelessWidget named ContactHeader to display Contact header. Put that widget in /ui/screens/contact_detail/widgets/contact_header.dart

File: lib/ui/screens/contact_detail_screen.dart [hunk 4]
```diff
@@ -1,20 +1,120 @@
...
+                            style: TextStyle(fontFamily: 'Peddana', fontSize: 25))
+                    ]
+                    )
+                ),
+                Spacer()
+            ]
+        ); 
+    }
+
+    List<Row> detailList(List<String> infoList){
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart]: Should create a dedicated StatelessWidget named ContactDetail to display Contact detail. Put that widget in /ui/screens/contact_detail/widgets/contact_detail.dart

File: lib/ui/screens/contact_detail_screen.dart [hunk 5]
```diff
@@ -1,20 +1,120 @@
...
     @override
     Widget build(BuildContext context) {
-        var contactName = contact.firstName + " " + contact.lastName + contact.id.toString();
+        //get full information of contact by id 
+        Contact contactInfo = MockContact.mockContact(contact.id); 
+        String contactName = contactInfo.firstName + " " + contactInfo.lastName + contactInfo.id.toString();
+        List<String> infoList = contactInfoList(contactInfo); 
+        String phone = contactInfo.phoneNumber; 
         return Scaffold(
-        appBar: AppBar(
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart:12]: Also try to set AppBar attribute to `backgroundColor: Colors.transparent,` and check the result.

File: lib/ui/screens/contact_detail_screen.dart [hunk 6]
```diff
@@ -1,20 +1,120 @@
...
+
     @override
     Widget build(BuildContext context) {
-        var contactName = contact.firstName + " " + contact.lastName + contact.id.toString();
+        //get full information of contact by id 
+        Contact contactInfo = MockContact.mockContact(contact.id); 
+        String contactName = contactInfo.firstName + " " + contactInfo.lastName + contactInfo.id.toString();
+        List<String> infoList = contactInfoList(contactInfo); 
+        String phone = contactInfo.phoneNumber; 
         return Scaffold(
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart:19]: Try the Scaffold with attribute `extendBodyBehindAppBar: true,` to get the full screen display effect.

File: lib/ui/screens/contact_list_screen.dart [hunk 7]
```diff
@@ -11,7 +11,7 @@ class ContactList extends StatefulWidget {
 
 class _ContactListState extends State<ContactList> { 
     //create some fake data and assign to contactList 
-    List<Contact> contactList = MockContact.mocks(); 
+    List<Contact> contactList = MockContact.mockList();
```
Comment (lngyeen) [inline lib/ui/screens/contact_list_screen.dart]: function name should be `mocks` which will return a List<> and `mock` which will return a single object.

File: lib/mockdata/mock_contact.dart [hunk 8]
```diff
@@ -3,15 +3,20 @@ class Contact{
...
+    Contact({required this.id, required this.firstName, required this.lastName, required this.phoneNumber, 
+            this.address = "", this.email =" ", this.gender = 0}); 
 
     @override 
     String toString() {
         return("id: $id\nFirst name: $firstName\nLast name: $lastName\nTel: $phoneNumber"); 
   }
 }
 class MockContact{
-    static List<Contact> mocks() =>
```
Comment (lngyeen) [inline lib/mockdata/mock_contact.dart:15]: function name should be mocks which will return a List<> and mock which will return a single object.

File: lib/ui/screens/contact_detail_screen.dart [hunk 9]
```diff
@@ -1,20 +1,120 @@
...
-        appBar: AppBar(
-            title: Text(contactName),
-        ),
-        body: Center(
-            
-        ),
+            appBar: AppBar(
+                title: Center(child: Column(
```
Comment (lngyeen) [inline lib/ui/screens/contact_detail_screen.dart]: Don't add too much information to the AppBar. Put the appBarPic in the body.

File: pubspec.yaml [hunk 10]
```diff
@@ -68,17 +67,23 @@ flutter:
...
-  #     fonts:
-  #       - asset: fonts/Schyler-Regular.ttf
-  #       - asset: fonts/Schyler-Italic.ttf
-  #         style: italic
-  #   - family: Trajan Pro
-  #     fonts:
-  #       - asset: fonts/TrajanPro.ttf
-  #       - asset: fonts/TrajanPro_Bold.ttf
-  #         weight: 700
+  fonts:
```
Comment (lngyeen) [inline pubspec.yaml:70]: It is not necessary to use custom fonts for this exercise.
The purpose of this exercise is to focus on layout and coding style. Don't focus too much on unnecessary elements. :)

**Review by lngyeen — CHANGES_REQUESTED** [review]
Please fix comments

File: lib/ui/screens/contact_detail_screen.dart (hunk 5 above)
Comment (heupotato) [reply lib/ui/screens/contact_detail_screen.dart:12]: Tried, but its result's not really suitable for my UI so I prefer not to do it :(

**Review by lngyeen — APPROVED** [review]

**Pull request merged**